docs/06/01/types.md

```diff
@@ -0,0 +1,159 @@
+
+```typescript
+export interface ComponentsContextType {
+  Alert: (props: AlertProps) => JSX.Element | null
+  Badge: (props: BadgeProps) => JSX.Element | null
+  Button: (props: ButtonProps) => JSX.Element | null
+  ButtonIcon: (props: ButtonIconProps) => JSX.Element | null
+  CalendarPreview: (props: CalendarPreviewProps) => JSX.Element | null
+  Card: (props: CardProps) => JSX.Element | null
+  Checkbox: (props: CheckboxProps) => JSX.Element | null
+  CheckboxGroup: (props: CheckboxGroupProps) => JSX.Element | null
+  ComboBox: (props: ComboBoxProps) => JSX.Element | null
+  DatePicker: (props: DatePickerProps) => JSX.Element | null
+  Heading: (props: HeadingProps) => JSX.Element | null
+  Link: (props: LinkProps) => JSX.Element | null
+  Menu: (props: MenuProps) => JSX.Element | null
+  NumberInput: (props: NumberInputProps) => JSX.Element | null
+  OrderedList: (props: OrderedListProps) => JSX.Element | null
+  PaginationControl?: (props: PaginationControlProps) => JSX.Element | null
+  ProgressBar: (props: ProgressBarProps) => JSX.Element | null
+  Radio: (props: RadioProps) => JSX.Element | null
+  RadioGroup: (props: RadioGroupProps) => JSX.Element | null
+  Select: (props: SelectProps) => JSX.Element | null
+  Switch: (props: SwitchProps) => JSX.Element | null
+  Table: <T>(props: TableProps<T>) => JSX.Element | null
+  Text: (props: TextProps) => JSX.Element | null
+  TextInput: (props: TextInputProps) => JSX.Element | null
+  UnorderedList: (props: UnorderedListProps) => JSX.Element | null
+}
+```
+
+Each component type (e.g., `ButtonProps`, `TextInputProps`) is defined in its respective file and exported from the SDK. These interfaces detail all the props that must be supported by your custom implementations.
+
+### GustoProviderCustomUIAdapterProps
+
+This interface defines the props for the `GustoProviderCustomUIAdapter` component ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/GustoProvider/GustoProviderCustomUIAdapter.tsx)):
+
+```typescript
+export interface GustoProviderCustomUIAdapterProps {
+  config: {
+    baseUrl: string
+    headers?: Record<string, string | number>
+  }
+  dictionary?: ResourceDictionary
+  lng?: string
+  locale?: string
+  currency?: string
+  theme?: DeepPartial<GTheme>
+  queryClient?: QueryClient
+  components: ComponentsContextType
+  children?: React.ReactNode
+}
+```
+
+The key props are:
+
+- `config`: Configuration for the API connection
+- `components`: Your custom component adapter implementing `ComponentsContextType`
+- `theme`: (Optional) Custom theming options
+- `dictionary`: (Optional) Translation overrides
+- `lng`: (Optional) Language setting
+- `locale` and `currency`: (Optional) Localization settings
+
+### Component Prop Types
+
+Each UI component has its own prop interface. Here are some of the most commonly used ones:
```

Comment:
I think this section could be omitted in favor of just referencing the component inventory which will be autogenerated after the work that i'm doing is implemented

Reply:
Updated